mypy/build.py

```diff
@@ -2342,11 +2342,15 @@ def dispatch(sources: List[BuildSource], manager: BuildManager) -> Graph:
         # after we loaded cache for whole graph. The `read_protocol_cache` will also validate
         # the protocol cache against the loaded individual cache files.
         TypeState.proto_deps = read_protocol_cache(manager, graph)
-        if TypeState.proto_deps is None and manager.stats.get('fresh_trees', 0) > 0:
+        if TypeState.proto_deps is None and manager.stats.get('fresh_metas', 0) > 0:
```

Comment:
It would be good to have a test case for this, since this is something that could easily get stale. Maybe create an issue about adding a test (maybe also for other related cache corruption issues)?

Reply:
We already have https://github.com/python/mypy/issues/5002, do we need a separate issue? I can just add a comment that we need to also check performance/behavior with corrupted caches.